sdk/inc/azure/iot/az_iot_hub_client.h

```diff
@@ -78,8 +77,10 @@ AZ_NODISCARD az_iot_hub_client_options az_iot_hub_client_options_default();
  * @param[in] iot_hub_hostname The IoT Hub Hostname.
  * @param[in] device_id The Device ID.
  * @param[in] options A reference to an #az_iot_hub_client_options structure. If `NULL` is passed,
- * `az_iot_hub_client_init()` will use the default options.
- * @return #az_result.
+ * az_iot_hub_client_init() will use the default options. If using custom options, please

```

Comment:
nit: I might be biased, but I like this wording better, especially because it doesn't repeat this same function name in the comment.
https://github.com/Azure/azure-sdk-for-c/blob/194e29567e7f8b215060186b9194f50c0d6ef60f/sdk/inc/azure/core/az_json.h#L323-L325

Reply:
This details what someone should do if they don't pass `NULL`. If they want custom options, they should first initialize the options with `az_iot_hub_client_options_default()` THEN put their options in. They shouldn't just allocate it, assign their options, and pass it to us.

Reply:
Let me clarify.

This is what I am suggesting:
```text
 * @param[in] options A reference to an #az_iot_hub_client_options structure. If `NULL` is passed,
 * the hub client will use the default options. If using custom options, please
 * initialize first by calling az_iot_hub_client_options_default() and then populating relevant
 * options with your own values.
```

That is, don't repeat the function name in the comment. Consider referring to `az_iot_hub_client_init()` as `the hub client`.

All the rest, including how to initialize and modify is all good.

Reply:
ah i see i see. good feedback will change